### PR TITLE
Pull Request for Issue #86 - Fix dependency vulnerabilities

### LINKED ISSUE #86: Remove marked security vulnerability
#### Issue Summary

Reported by GitHub:

Known moderate severity security vulnerability detected in marked < 0.3.7 defined in package-lock.json.

package-lock.json update suggested: marked ~> 0.3.7.

### PR DESCRIPTION
**Pull request by robincher**
Fixes #86 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation about the functionality in the appropriate .md file
- [ ] I have added in line documentation to the code I modified

### Short description of what this PR does:

- Update esdoc dev dependency version 
- Remove away marked (sub-dependency of esdoc) vulnerability 
- Bump package.json version by minor increment
- Update License End-Year based on existing test requirement

### CODE REVIEW

**Comment by SendGridDX** [comment]
[CLA assistant check](https://cla.sendgrid.com/sendgrid/nodejs-http-client?pullRequest=87) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.sendgrid.com/sendgrid/nodejs-http-client?pullRequest=87) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.sendgrid.com/check/sendgrid/nodejs-http-client?pullRequest=87) it.</sub>

**Comment by robincher** [comment]
Additional vulnerabilities was flagged out by running ```npm audit```, and it was suggested to bump up mocha version to 5.x.x.  Should i fix that within this Pull request too?

**Comment by thinkingserious** [comment]
This is awesome @robincher!

I've added this to our backlog for a review and merge.

With Best Regards,

Elmer

File: package.json [hunk 1]
```diff
@@ -5,7 +5,7 @@
   ],
   "name": "sendgrid-rest",
   "description": "HTTP REST client, simplified for Node.js.",
-  "version": "2.4.0",
+  "version": "2.4.1",
```
Comment (thinkingserious) [inline package.json]: Please set this back to 2.4.0. I will be updating the version number at the next release. Thanks!

Comment (robincher) [reply package.json]: Got it, will commit the corresponding package-lock.json too

**Comment by robincher** [comment]
Hi @thinkingserious , additionally i updated the dev dependency - mocha to the latest. The previous version 2.x.x was having vulnerabilities if you run ```npm audit```. I have run the local test and its all good

**Comment by thinkingserious** [comment]
Note to code reviewers: we need to make sure none of these updates are breaking changes.

**Comment by thinkingserious** [comment]
Hello @robincher,<br /><br/>Thanks again for the PR!<br />                                              <br />It's <a href="https://sendgrid.com/blog/hacktoberfest-2018-has-arrived/">HACKTOBERFEST</a>! We want to show our appreciation by sending                                              you some special Hacktoberfest swag. If you have not already, could you please fill out                                              <a href="https://dx.sendgrid.com/hacktoberfest/join">this form</a> so we can send it to you? Thanks!                                             <br /><br />Team SendGrid DX

**Comment by robincher** [comment]
Opps sorry @thinkingserious , i have a separate PR (incoming), so i have reverted the wrong commits. Need you to resign the CLA again :(

**Comment by thinkingserious** [comment]
Closing in favor of https://github.com/sendgrid/sendgrid-nodejs/pull/1010

**Pull request closed without merging**